### PR TITLE
Add: option to view plaintext chat message. Issue/2508

### PR DESCRIPTION
**Pull request by CertifiedJoon**
Fixes: #2508 

Rendered by suspense
image

Plain text
image

I am not sure whether using memo on isPlainText has any performance gain. I need your opinions.

### CODE REVIEW

File: website/src/components/Chat/ChatMessageEntry.tsx [hunk 1]
```diff
@@ -92,7 +92,9 @@ export const ChatMessageEntry = memo(function ChatMessageEntry({
 
   const isAssistant = message.role === "assistant";
   const [isEditing, setIsEditing] = useBoolean(false);
+  const [isPlainText, setIsPlainText] = useState<boolean>(false);
   const inputRef = useRef<HTMLTextAreaElement>(null);
+  const memoIsPlainText = useMemo(() => isPlainText, [isPlainText]);
```
Comment (notmd) [inline website/src/components/Chat/ChatMessageEntry.tsx:97]: This probably has no effect, you can remove this.

Comment (CertifiedJoon) [reply website/src/components/Chat/ChatMessageEntry.tsx]: noted. i have made the change.

File: website/src/components/Chat/ChatMessageEntry.tsx [hunk 2]
```diff
@@ -175,6 +177,12 @@ export const ChatMessageEntry = memo(function ChatMessageEntry({
                 )}
                 {state === "complete" && (
                   <>
+                    <BaseMessageEmojiButton
```
Comment (notmd) [inline website/src/components/Chat/ChatMessageEntry.tsx:180]: Can you adjust icon base on the `isPlainText` state? I think using these two icons will make more sense
 https://tabler-icons.io/i/markdown-off
 https://tabler-icons.io/i/markdown

Comment (CertifiedJoon) [reply website/src/components/Chat/ChatMessageEntry.tsx:222]: noted. have made the relevant changes.

**Review by notmd — APPROVED** [review]
Thank you!

**Pull request merged**